sdk/communication/azure-communication-identity/src/main/java/com/azure/communication/identity/CommunicationIdentityClient.java

```diff
@@ -60,13 +61,14 @@ public CommunicationUserIdentifier createUser() {
         Response<CommunicationIdentityAccessTokenResult> response = 
             client.createWithResponseAsync(new CommunicationIdentityCreateRequest(), context).block();
         
-        if (response != null && response.getValue() != null) {
-            String id = response.getValue().getIdentity().getId();
-            return new SimpleResponse<CommunicationUserIdentifier>(
-                response,
-                new CommunicationUserIdentifier(id));
+        if (response == null || response.getValue() == null) {
+            throw logger.logExceptionAsError(new IllegalStateException("Create user response and value cannot be null"));
```

Comment:
Better wording. Response and value are returned from service not coming from caller. We can't blame the caller here. :-)

E.g. Service failed to return a response or expected value.